src/3k-mapper.xml

```diff
@@ -1154,6 +1169,170 @@ end</script>
                     <packageName></packageName>
                     <regex>^tmap addalias &quot;(.+)&quot; (.+)$</regex>
                 </Alias>
+                <Alias isActive="yes" isFolder="no">
+                    <name>Rename an area</name>
+                    <script>--renames current area
+local Name = matches[2]
+local cArea = getRoomArea(tmap.lastId)
+setAreaName( cArea, Name )</script>
+                    <command></command>
+                    <packageName></packageName>
+                    <regex>^tmap renamearea (.+)$</regex>
+                </Alias>
+                <Alias isActive="yes" isFolder="no">
+                    <name>Connect all stub in area</name>
+                    <script>local aid = getRoomArea(tmap.lastId)
+local rooms = getAreaRooms(aid)
+for i,v in pairs(rooms) do
+	local stubs = getExitStubs(v)
+	if stubs and type(stubs) == 'table' then
+		for j,k in pairs(stubs) do
+			connectExitStub(v,k)
+		end
+	end
+end</script>
+                    <command></command>
+                    <packageName></packageName>
+                    <regex>^tmap connectstubs$</regex>
+                </Alias>
+                <Alias isActive="yes" isFolder="no">
+                    <name>Add room speedwalk</name>
+                    <script>if not tmap.mwalkList then
+	tmap.mwalkList = {}
+end
+tmap.mwalkList[matches[2]] = tmap.lastId
+tmap:echo(&quot;Added &quot;..matches[2]..&quot;!&quot;)</script>
+                    <command></command>
+                    <packageName></packageName>
+                    <regex>^tmap addwalk (.+)$</regex>
+                </Alias>
+                <Alias isActive="yes" isFolder="no">
+                    <name>Delete Area</name>
+                    <script>if not matches[2] then
+	tmap:echo(&quot;Syntax: tmap deleteArea area name&quot;)
+else
+	local atable = getAreaTableSwap()
```

Comment:
`TLuaInterpreter::deleteArea(...)` is now fully working and will take an existing area name as a string to delete.  The code here is, I think, redundant.

:beetle: However, there is a nasty corner case in that `deleteArea(...)` called with the same string as `TRoomDB::mDefaultAreaName` may delete the `-1` area (I'll have to check more deeply), which does not sound good! :scream: